api-endpoints/user-profile/user-profile.interface.ts

```diff
@@ -1,6 +1,7 @@
 interface Language {
     name: string;
     level: string;
+    _id?:string | any
```

Comment:
the reason why this is like this is because the _id is not available initially when creating it so its from the return we are getting it and if I don't add it here it as type of any it will flag serious error

Reply:
You're already marking the `_id ` as optional. The `any` wasn't necessary.
You could omit the `_id ` when trying to create*

```suggestion
    interface Language {
    name: string;
    level: string;
    _id: string
}

type OmitId = Omit<Language, '_id'>
```